lab-spencer/lib/storage.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+let uuid = require('uuid');
+let fs = require('fs-extra');
+
+let storage = module.exports = {};
+
+storage.setItem = data => {
+  data.id = uuid.v1();
+  return fs.writeJson(`${__dirname}/../data/${data.id}`, data)
+    .then(() => Promise.resolve(data));
+};
+
+storage.fetchItem = id => {
```

Comment:
This function has a bit of an antipattern in it. Since you check for the error twice, it won't be a bug though.

If you hadn't checked for the error again on line 20, you might have run in to a scenerio like this.

 - You check to see if a file exists, and it does.
 - Some other code deletes the file, this could be another process on your computer, or even another request to your server.
 - Your call to `readJson` happens and you assume it exists, even though it doesn't
 - Errors happen, sad faces are worn by all.

TL;DR, race conditions are a bitch. When you do IO, just try to do the thing you want to do, and handle the errors if they happen.